NuGet/NuGetGallery
# Phase 2 of SemVer2 support

This PR is to combine all semver2 related work into a single feature branch PR, part of phase 2 deployment.

Verified in DEV:
* Remove validation rules that block uploading valid semver2 versions #3645 (PR #3757)
* Add support for semVerLevel query parameter to V2 endpoints #3557 (PR #3714)
* Maintain legacy 2.x client compatibility in non-semver2 validation logic #3645 (PR #3761)
* V2 NuGetEntityTypeSerializer Id link patcher must retain curated feed name #3861 (PR #3864)
* Index Recommendation for SemVerLevelKey on NuGetGallery [#435](https://github.com/NuGet/Engineering/issues/435) (PR #3867)
* Support IsLatest and IsAbsoluteLatest with respect to SemVer 2.0.0 in gallery DB #3708 (PR #3842)
* absoluteLatest endpoint on view needs to be updated #3751 (PR #3842)
* LuceneIndexingService in Gallery should take into account IsLatest(Stable)SemVer2 #3862 (PR #3863)
* Add unit tests for NuGetEntityTypeSerializer Id link patcher #3871 (PR #3879)
* "Package published" notification email should use NormalizedVersion in URL #3885 (PR #3886)
* Fix Parsing of Empty Dependency Groups #3890
* Show full version on package details page #3870 (PR #3887)
* Fix bug in IsLatest(Stable)SemVer2 #3895
* Highlight semver-level on package details page #3796 (PR #3893)
* Add end-to-end tests that use nuget.exe to perform SemVer 2.0.0 operations #3788
* VerifyPackage on ApiController should treat version as optional parameter #3903 
* UpdateIsLatest not resetting IsLatest(Stable)SemVer2 on previous latest versions #3908 (PR #3909)
* Weird flow when uploading semver2 package through Gallery #3912 (PR #3915)
* Default to latest stable semver2 on package details page PR #3930
* When unlisting package with metadata 404 page shown after ok click #3907 (PR #3925)
* User profile page does not show SemVer 2.0.0 packages #3911 (PR #3933)
* Fix functional test failure SearchMicrosoftDotNetCuratedFeed #3941 (PR #3942)
* Fix System.NotSupportedException on User profile page #3940 (PR #3943)
* Fix Functional Test failure for ODataFeeds.V2FeedExtendedTests.FindPackagesByIdTest #3947 (PR #3948)
* Fix load test failure due to incorrect test setup #3956 (PR #3957)
* $filter=IsLatestVersion and semVerLevel=2.0.0 yields no results #3905 (PR #3966)
* Detect if package only differ by metadata and show optimal user-facing error message #3910 (PR #3970)
* Update Semver2 package details message with final nuget client version #3897  (PR #3988)
* On package validation failure an actionable error message should be displayed. #3916 (PR #4031)

Once merged, also merge this:
* Build script for NuGet.Shims does not compile when no VS2015 available https://github.com/NuGet/Engineering/issues/433
